service/history/task/cross_cluster_task.go

```diff
@@ -35,14 +37,45 @@ import (
 	"github.com/uber/cadence/common/persistence"
 	ctask "github.com/uber/cadence/common/task"
 	"github.com/uber/cadence/common/types"
+	"github.com/uber/cadence/service/history/execution"
 	"github.com/uber/cadence/service/history/shard"
 )
 
 // cross cluster task state
+// a typically state transition will be:
+// Initialized -> Reported -> Recorded -> Reported
+// if task has two stages (e.g. first start childworkflow and
+// then schedule first decision task)
+// or
+// Initialized -> Reported
+// if task has only one stage (e.g. cancel external workflow)
+//
+// NOTE: DO NOT change the value for each state
+// new states must be added at the end to ensure backward compatibility
+// across clusters.
 const (
-	processingStateInitialized processingState = iota + 1
-	processingStateResponseReported
-	processingStateResponseRecorded
+	// processingStateInitialized is the initial state after a task is loaded
+	// task is available for poll at this state
+	processingStateInitialized processingState = 1
```

Comment:
Probably ALL_CAPS would be a better const naming.

Reply:
I think ALL_CAPS is against golang naming convention and will result in a lint warning like the following.
```
service/history/task/cross_cluster_task.go:59:2: [var-naming] don't use ALL_CAPS in Go names; use CamelCase
```
so will keep the existing naming.